task_sdk/tests/execution_time/test_supervisor.py

```diff
@@ -258,6 +261,55 @@ def test_run_simple_dag(self, test_dags_dir, captured_logs, time_machine):
             "timestamp": "2024-11-07T12:34:56.078901Z",
         } in captured_logs
 
+    def test_state_conflict_on_heartbeat(self, captured_logs, monkeypatch, mocker):
+        """
+        Test that ensures that the Supervisor does not cause the task to fail if the Task Instance is no longer
+        in the running state.
+        """
+        structlog.configure(wrapper_class=structlog.make_filtering_bound_logger(logging.ERROR))
```

Comment:
We should probably make this an autouse fixture in this class since we have it in every test.

Reply:
moved it to `captured_logs` fixture fixed in https://github.com/apache/airflow/pull/44406/commits/6ed56ce1f987990b1adbc1172c2de357e45a2413